### PR TITLE
feat: Api endpoint roadworks

### PR DESCRIPTION
**Pull request by alainajaffer**
some tests not working from parent branch

### CODE REVIEW

File: packages/api/client.ts [hunk 1]
```diff
@@ -801,3 +801,84 @@ export const getAdminAreas = async (dbClient: Kysely<Database>) => {
...
+            "roadworks.proposedStartDateTime",
+            "roadworks.proposedEndDateTime",
+            "roadworks.actualStartDateTime",
+            "roadworks.actualEndDateTime",
+            "roadworks.permitStatus",
+            "roadworks.workStatus",
+            "highway_authority_admin_areas.administrativeAreaCode",
+        ])
+        .distinct()
+        .orderBy("roadworks.permitReferenceNumber")
```
Comment (ljones92) [inline packages/api/client.ts]: Ordering by proposedStartDateTime might make more sense here

File: packages/api/client.ts [hunk 2]
```diff
@@ -801,3 +801,84 @@ export const getAdminAreas = async (dbClient: Kysely<Database>) => {
...
+            "roadworks.proposedEndDateTime",
+            "roadworks.actualStartDateTime",
+            "roadworks.actualEndDateTime",
+            "roadworks.permitStatus",
+            "roadworks.workStatus",
+            "highway_authority_admin_areas.administrativeAreaCode",
+        ])
+        .distinct()
+        .limit(1)
+        .execute();
```
Comment (ljones92) [inline packages/api/client.ts]: We can use `executeTakeFirst()` instead here and get rid of the limit

File: packages/api/client.ts (hunk 1 above)
Comment (alainajaffer) [reply packages/api/client.ts]: done

File: packages/api/client.ts (hunk 2 above)
Comment (alainajaffer) [reply packages/api/client.ts]: done

File: stacks/Api.ts [hunk 3]
```diff
@@ -201,6 +233,8 @@ export function ApiStack({ stack }: StackContext) {
             "GET /area-codes": areaCodeFunction,
             "GET /admin-areas": adminAreasFunction,
             "POST /street-manager": postStreetManagerFunction,
+            "GET /roadworks": roadworksFunction,
+            "GET /roadwork/{pathReferenceNumber}": roadworkByIdFunction,
```
Comment (DeannaSharma) [inline stacks/Api.ts]: I think this variable should be `permitReferenceNumber`

Comment (DeannaSharma) [reply stacks/Api.ts]: updated and pushed

File: stacks/Api.ts [hunk 4]
```diff
@@ -201,6 +233,8 @@ export function ApiStack({ stack }: StackContext) {
             "GET /area-codes": areaCodeFunction,
             "GET /admin-areas": adminAreasFunction,
             "POST /street-manager": postStreetManagerFunction,
+            "GET /roadworks": roadworksFunction,
+            "GET /roadwork/{permitReferenceNumber}": roadworkByIdFunction,
```
Comment (ljones92) [inline stacks/Api.ts]:
```suggestion
            "GET /roadworks/{permitReferenceNumber}": roadworkByIdFunction,
```
Keep it as roadworks to stay inline with the RESTful approach

Comment (DeannaSharma) [reply stacks/Api.ts]: done

**Review by ljones92 — APPROVED** [review]

**Pull request merged**